### PR TITLE
LPS-123739 [7.2, 7.3, master] Fix inability to hide the preferred language message during the user session

### PR DESCRIPTION
**Pull request by jcampoy**
**Description of the issue**

If the user discards the message that the portal shows when the user's preferred language differs from the language in the url, the action is not persisted during his http session. As a result, the user sees the message in each request.

**Steps to reproduce:**

1. Sign-in as Test
2. Verify that the user's preferred language is English, and also it is default language of the site.
3. Go to the _localhost:8080/web/guest/home_ page 
4. Go to _localhost:8080/es/web/guest/home_ => User will be notified through a toast on the bottom left that can swap to English version or change language preferences.
5. Close the message by clicking the X icon on the toast.
6. Reload localhost:8080/es/web/guest/home

**Expected result**

During the session, the user's action is remembered and he does not view the message again.

**Result obtained** 

The message is displayed again. In fact, every time you move to a different page, that toast will appear again and again.

**Solution**

Based on how it works in 7.1 and before (see [here](https://github.com/liferay/liferay-portal-ee/blob/7.1.x/portal-web/docroot/html/common/themes/user_locale_options.jsp#L52-L62) and [here](https://github.com/liferay/liferay-portal-ee/blob/7.1.x/modules/apps/frontend-js/frontend-js-aui-web/src/main/resources/META-INF/resources/liferay/store.js)) we are storing the user's action in the HttpSession again. 

Pd: Last commit is just trying to simplify and avoid duplicated code. Feel free to get rid of it in case it does not match our frontend guidelines.

Previous iterations of this PR:

- https://github.com/liferay-frontend/liferay-portal/pull/540
- https://github.com/liferay-frontend/liferay-portal/pull/560
- https://github.com/liferay-frontend/liferay-portal/pull/569

last try @jbalsas, be free to apply any changes you considere :)

### CODE REVIEW

**Comment by jcampoy** [comment]
ci:test

File: modules/apps/frontend-js/frontend-js-web/src/main/resources/META-INF/resources/liferay/util/session.es.js [hunk 1]
```diff
@@ -38,14 +38,19 @@ function getSessionClickURL() {
 /**
  * Gets the Store utility fetch value for given key
  * @param {String} key string for fetch request
+ * @param {Array} List of options
```
Comment (wincent) [inline modules/apps/frontend-js/frontend-js-web/src/main/resources/META-INF/resources/liferay/util/session.es.js:41]: It's not an array or a list, so we should amend this (same on line 74).

**Review by wincent — COMMENTED** [review]
Minor nit noted inline, I'm just going to pull this down and update it, so don't worry about it @jcampoy.

**Comment by wincent** [comment]
ci:test:sf

**Comment by wincent** [comment]
ci:test:relevant

**Comment by jbalsas** [comment]
Thanks @jcampoy and @wincent for getting this through!! 😘

**Comment by jbalsas** [comment]
ci:forward

**Pull request closed without merging**